deepcell/datasets/__init__.py

```diff
@@ -42,128 +45,80 @@ class Dataset:
     """
 
     def __init__(self,
-                 path,
+                 basename,
                  url,
                  file_hash,
-                 metadata):
-        self.path = path
+                 secure=False):
+        self.data_dir = os.path.expanduser(os.path.join('~', '.keras', 'datasets'))
+        self.path = os.path.join(self.data_dir, basename)
         self.url = url
+        self.secure = secure
         self.file_hash = file_hash
-        self.metadata = metadata
 
-    def _load_data(self, path, mode, test_size=0.2, seed=0):
-        """Loads dataset.
+    def _get_data(self):
+        if not os.path.exists(self.data_dir):
+            os.makedirs(self.data_dir)
+        elif not os.path.isdir(self.data_dir):
+            raise OSError(f'{self.data_dir} exists but is not a directory')
 
-        Args:
-            test_size (float): fraction of data to reserve as test data
-            seed (int): the seed for randomly shuffling the dataset
+        if self.secure:
+            fetch_data()
+        else:
+            path = get_file(self.path, origin=self.url, file_hash=self.file_hash)
 
-        Returns:
-            tuple: (x_train, y_train), (x_test, y_test).
-        """
-        basepath = os.path.expanduser(os.path.join('~', '.keras', 'datasets'))
-        prefix = path.split(os.path.sep)[:-1]
-        data_dir = os.path.join(basepath, *prefix) if prefix else basepath
-        if not os.path.exists(data_dir):
-            os.makedirs(data_dir)
-        elif not os.path.isdir(data_dir):
-            raise OSError(f'{data_dir} exists but is not a directory')
-
-        path = get_file(path,
-                        origin=self.url,
-                        file_hash=self.file_hash)
-
-        train_dict, test_dict = get_data(
-            path,
-            mode=mode,
-            test_size=test_size,
-            seed=seed)
-
-        x_train, y_train = train_dict['X'], train_dict['y']
-        x_test, y_test = test_dict['X'], test_dict['y']
-        return (x_train, y_train), (x_test, y_test)
-
-    def load_data(self, path=None, test_size=0.2, seed=0):
-        """Loads dataset.
+    @abc.abstractmethod
+    def load_data(self, split='val'):
+        """Load dataset from specified split
 
         Args:
-            path (str): filepath to save the data locally.
-            test_size (float): fraction of data to reserve as test data
-            seed (int): the seed for randomly shuffling the dataset
-
-        Returns:
-            tuple: (x_train, y_train), (x_test, y_test).
+            split (str, optional): Dataset split, one of 'train', 'test', 'val'. Defaults to 'val'.
         """
-        path = path if path else self.path
-        return self._load_data(path, 'sample', test_size=test_size, seed=seed)
+        raise NotImplementedError
 
-    def load_tracked_data(self, path=None, test_size=0.2, seed=0):
-        """Loads dataset using "siamese_daughters" mode.
+    def load_license(self):
```

Comment:
If we are always going to use the same apache license for datasets that we use for code, we could just point to http://www.github.com/vanvalenlab/deepcell-tf/LICENSE in the docstring instead of packaging a license file with each dataset.